rdrf/registry/trials/models.py

```diff
@@ -0,0 +1,98 @@
+from django.db import models
+
+from rdrf.models.definition.models import Registry
+from registry.patients.models import Patient
+
+
+class NofOneTrial(models.Model):
+    title = models.CharField(max_length=30)
+    description = models.TextField(blank=True)
+    registry = models.ForeignKey(Registry, on_delete=models.CASCADE, related_name="nofone_trials")
+
+    class Meta:
+        verbose_name = "N-of-1 Trial"
+        verbose_name_plural = "N-of-1 Trials"
+
+    def __str__(self):
+        return f"{self.title} ({self.registry.code})"
+
+
+class NofOneTreatment(models.Model):
+    title = models.CharField(max_length=100)
+    blinded_title = models.CharField(max_length=20)
+    description = models.TextField(blank=True)
+    trial = models.ForeignKey(NofOneTrial, on_delete=models.CASCADE, related_name="trials")
+
+    class Meta:
+        verbose_name = "N-of-1 Treatment"
+        verbose_name_plural = "N-of-1 Treatments"
+
+    def __str__(self):
+        return f"{self.blinded_title} ({self.title})"
+
+
+class NofOneArm(models.Model):
+    patient = models.ForeignKey(Patient, on_delete=models.PROTECT, blank=True, null=True)
+    trial = models.ForeignKey(NofOneTrial, on_delete=models.PROTECT, related_name="arms")
+
+    sequence_index = models.IntegerField()
+
+    class Meta:
+        verbose_name = "N-of-1 Arm"
+        verbose_name_plural = "N-of-1 Arms"
+
+    @property
+    def ordered_periods(self):
+        return self.cycles.select_related("periods").all().order_by("start")
+
+    def __str__(self):
+        return f"{self.patient} - {self.trial}"
+
+
+class NofOneCycle(models.Model):
+    arm = models.ForeignKey(NofOneArm, on_delete=models.PROTECT, related_name="cycles")
+
+    sequence_index = models.IntegerField()
```

Comment:
I would rather order by datetime than `sequence_index`, but because patients don't all begin the trial at once, there isn't a definitive start time.